refactor(tooltip): follow current style guides and fix typescript issues

components is not currently following our current style guides and tsc is returning following errors

```
21       'bl-tooltip': TooltipProps & Partial<HTMLElement>,
                       ~~~~~~~~~~~~

Tooltip.tsx(21,21): error TS4033: Property ''bl-tooltip'' of exported interface has or is using private name 'TooltipProps'.


26 export class Tooltip extends Component<TooltipProps> {
                                          ~~~~~~~~~~~~

Tooltip.tsx(26,40): error TS4020: Extends clause of exported class 'Tooltip' has or is using private name 'TooltipProps'.


28   static get props() {
                ~~~~~

Tooltip.tsx(28,14): error TS4038: Return type of public static property getter from exported class has or is using name 'PropOptions' from external module "/Users/mario/ngParty/blaze-elements/node_modules/skatejs/src/ts-typings/api" but cannot be named.
```

--- a/packages/tooltip/Tooltip.tsx
+++ b/packages/tooltip/Tooltip.tsx
@@ -1,48 +1,49 @@
-import { h, Component, prop } from 'skatejs';
+import { h, Component } from 'skatejs';
 import styles from './Tooltip.scss';
-import { css } from '@blaze-elements/common';
+import { css, prop, shadyCssStyles } from '@blaze-elements/common';
 
-const TooltipTypes = {
+export const TooltipTypes = {
   top: 'top',
   left: 'left',
   right: 'right',
   bottom: 'bottom',
 };
 
-// public
-type TooltipProps = {
+export type TooltipProps = Props & EventProps;
+
+export type Attrs = {
   type?: keyof typeof TooltipTypes,
-  label: string,
+  label?: string,
 };
 
-declare global {
-  namespace JSX {
-    interface IntrinsicElements {
-      'bl-tooltip': TooltipProps & Partial<HTMLElement>,
+export type Props = {
+  type?: keyof typeof TooltipTypes,
+  label?: string,
+};
+
+export type EventProps = {};
+
+export type Events = {};
+
+@shadyCssStyles()
+export default class Tooltip extends Component<TooltipProps> {
+
+  @prop( {
+    type: String,
+    attribute: {
+      source: true
+    },
+    default: 'right'
+  } ) type: keyof typeof TooltipTypes;
+
+  @prop( {
+    type: String,
+    attribute: {
+      source: true
     }
-  }
-};
+  } ) label: string;
 
-export class Tooltip extends Component<TooltipProps> {
-  static get is() { return 'bl-tooltip'; }
-  static get props() {
-    return {
-      type: prop.string( {
-        attribute: {
-          source: true
-        },
-        default: 'right'
-      } ),
-      label: prop.string( {
-        attribute: {
-          source: true
-        }
-      } )
-    };
-  }
-
-  type: string;
-  label: string;
+  get css() { return styles; }
 
   renderCallback() {
     const { label, type } = this;
@@ -55,15 +56,14 @@ export class Tooltip extends Component<TooltipProps> {
       }
     );
 
-    return [
-      <style>{styles}</style>,
+    return (
       <span
         className={className}
         aria-label={label}
       >
         <slot />
       </span>
-    ];
+    );
   }
 
 }

--- a/packages/tooltip/index.demo.tsx
+++ b/packages/tooltip/index.demo.tsx
@@ -1,12 +1,12 @@
 import { h, Component } from 'skatejs';
 import { Tooltip } from './index';
+import { customElement } from '@blaze-elements/common';
 
+@customElement( 'bl-tooltip-demo' )
 export class Demo extends Component<void> {
-  static get is() { return 'bl-tooltip-demo'; }
 
   renderCallback() {
-    return [
-      <style />,
+    return (
       <fieldset>
         <legend>{Tooltip.is}</legend>
 
@@ -18,8 +18,6 @@ export class Demo extends Component<void> {
         </div>
 
       </fieldset>
-    ];
+    );
   }
 }
-
-customElements.define( Demo.is, Demo );

--- a/packages/tooltip/index.ts
+++ b/packages/tooltip/index.ts
@@ -1,9 +1,19 @@
-import { define } from 'skatejs';
 
-import { Tooltip } from './Tooltip';
+import { customElement, GenericTypes } from '@blaze-elements/common';
+import RawTooltip, { TooltipProps, Attrs, Events } from './Tooltip';
 
-// export for public access to Skate Class
-export { Tooltip } from './Tooltip';
+const Tooltip = customElement( 'bl-tooltip' )( RawTooltip ) as typeof RawTooltip;
 
-// register WebComponent by our name
-define( Tooltip );
+export {
+  Tooltip
+};
+
+
+declare global {
+  namespace JSX {
+    interface IntrinsicElements {
+      'bl-tooltip': GenericTypes.IntrinsicCustomElement<TooltipProps>
+      & GenericTypes.IntrinsicBoreElement<Attrs, Events>
+    }
+  }
+}

